pkg/cmd/browse/browse.go

```diff
@@ -0,0 +1,245 @@
+package browse
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/spf13/cobra"
+)
+
+type browser interface {
+	Browse(string) error
+}
+
+type BrowseOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+	Browser    browser
+
+	SelectorArg   string
+	AdditionalArg string
+
+	ProjectsFlag bool
+	WikiFlag     bool
+	SettingsFlag bool
+	BranchFlag   bool
+}
+
+type exitCode int
+
+const (
+	exitUrlSuccess    exitCode = 0
+	exitNonUrlSuccess exitCode = 1
+	exitNotInRepo     exitCode = 2
+	exitTooManyFlags  exitCode = 3
+	exitTooManyArgs   exitCode = 4
+	exitExpectedArg   exitCode = 5
+	exitInvalidCombo  exitCode = 6
+)
+
+func NewCmdBrowse(f *cmdutil.Factory) *cobra.Command {
+
+	opts := &BrowseOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		Browser:    f.Browser,
+		BaseRepo:   f.BaseRepo,
+	}
+
+	cmd := &cobra.Command{
+		Long:  "Open specific pages in the browser on GitHub",
+		Short: "Open GitHub in the browser",
+		Use:   "browse {<number> | <path>}",
+		Args:  cobra.RangeArgs(0, 2),
+		Example: heredoc.Doc(`
+			$ gh browse
+			#=> Open current repository to the default branch
+
+			$ gh browse 217
+			#=> Open issue or pull request 217
+
+			$ gh browse --settings
+			#=> Open repository settings
+
+			$ gh browse main.go:312
+			#=> Open main.go at line 312
+
+			$ gh browse main.go --branch main
+			#=> Open main.go in the main branch
+		`),
+		Annotations: map[string]string{
+			"IsCore": "true",
+			"help:arguments": heredoc.Doc(`
+				A browser location can be specified using arguments in the following format:
+				- by number for issue or pull request, e.g. "123"; or
+				- by path for opening folders and files, e.g. "cmd/gh/main.go"
+			`),
+			"help:environment": heredoc.Doc(`
+				To configure a web browser other than the default, use the BROWSER environment variable 
+			`),
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			if len(args) > 1 {
+				opts.AdditionalArg = args[1]
```

Comment:
What does `AdditionalArg` represent? As far as I understand, there is only 1 optional argument allowed for the `browse` command and that is already saved into `opts.SelectorArg`, right?

Reply:
The `AdditionalArg` is for the `BranchFlag`, because `browse --branch` can take one or two args. As I understand it from vilmibm we should use `StringVarP` to avoid the mess of `AdditionalArg` and `BranchFlag`. We're going to try sorting that out today.